src/files-rename.py

```diff
@@ -0,0 +1,24 @@
+import ctypes
+import os
+import sys
+
+AT_FDCWD = -100
+
+RENAME_NOREPLACE = (1 << 0)
+# RENAME_EXCHANGE = (1 << 1)
+
+libc6 = ctypes.CDLL(None, use_errno=True)
+
+if not libc6.renameat2:
+    sys.stderr.write("renameat2 not available")
+    sys.exit(1)
+
+source = sys.argv[1]
+dest = sys.argv[2]
+# overwrite = sys.argv[3]
+
+if libc6.renameat2(AT_FDCWD, source.encode(), AT_FDCWD, dest.encode(), RENAME_NOREPLACE) == -1:
```

Comment:
`encode()` isn't quite right here.  To match what Python normally does with filesystem encoding (which is what the bridge does with the filenames in fsinfo) you need to use surrogate escapes.

Reply:
It's actually kinda funny since the string we have in `sys.argv` was actually decoded from the commandline (after being encoded for it, from a string, by the bridge).

In all of those cases, though, I think `errors='surrogateescape'` is what's in play.

Reply:
Thanks! I kinda wish we could spawn with native types auto converted like dbus but eh :-) (dream on, dreaaam on)